### PR TITLE
First check in

### PR DESCRIPTION
**Pull request by BrynjarS**
Documentation of reporting tables

### CODE REVIEW

File: source/reporting/Report database.rst [hunk 1]
```diff
@@ -0,0 +1,248 @@
...
+Special fields in SubscriberContacts:
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+**IsPrimary**: Value 1 means the address is home address, 0 is billing address or temporary address.
+
+**Identification**: May contain information about organization number, customer reference and more. JSON functions must be used to retrieve each individual value in this type of field.
+ 
+
+[dbo].[Subscribers]
+-------------------
+Subscriber is a register of all subscribers and contains all contact information. This information is used for deliveries and invoicing. Examples of use are sending a PDF invoice to an e-mail address or sending a physical invoice to a postal address.
```
Comment (havmunke) [inline source/reporting/Report database.rst:48]: Important that the contact information here is fetched from the primary address.

File: source/reporting/Report database.rst [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+**ContactId**: This is a reference to the subscriber contact from which the contact information is retrieved.
+
+**ExternalId**: External customer number of a third party system.
+
+ 
+[DBO].[Tag]
+---------
+Free text that can be sent with an order, but this is used differently from customer to customer. The idea is to be able to connect to other tables to be able to retrieve information. Information in Tags can also be created manually.
+
+When you send a tag with an order, two tags are currently created in the report layer. One with type Subscriber and one with type SubscriberAccount. Referenceid then correspond henholdsvi s SubscriberId and SubscriberAccount . Thus you can search for all subscribers who have a relationship with for example ShopID = 458. The vast majority of tables in the report layer have a connection to subscriberide and can then be grouped / filtered in relation to Tag.
```
Comment (havmunke) [inline source/reporting/Report database.rst:61]: A little bit norwegian here...

File: source/reporting/Report database.rst [hunk 3]
```diff
@@ -0,0 +1,248 @@
+ReportingService: Documentation of tables
```
Comment (esbenbach) [inline source/reporting/Report database.rst:1]: I think we should avoid the use of "ReportingService" and instead talk about it as the Reporting SQL Database (or just Reporting Database).

No point in giving external parties a headache about the fact its wrapped in its own service :)

File: source/reporting/Report database.rst [hunk 4]
```diff
@@ -0,0 +1,248 @@
...
+[DBO].[Tag]
+---------
+Free text that can be sent with an order, but this is used differently from customer to customer. The idea is to be able to connect to other tables to be able to retrieve information. Information in Tags can also be created manually.
+
+When you send a tag with an order, two tags are currently created in the report layer. One with type Subscriber and one with type SubscriberAccount. Referenceid then correspond henholdsvi s SubscriberId and SubscriberAccount . Thus you can search for all subscribers who have a relationship with for example ShopID = 458. The vast majority of tables in the report layer have a connection to subscriberide and can then be grouped / filtered in relation to Tag.
+ 
+
+[economy].[AccountTransaction]
+----------------------------
+Saldo transactions can, among other things, be used for purchases from agreements. For example, if you agree on a payment claim, which you will later buy out of, then this table will be used.
```
Comment (havmunke) [inline source/reporting/Report database.rst:66]: I think it's better described as balance-transactions of different types. Will be created if paid too much or too litle, if a subscription is cancelled or for imported transactions.

File: source/reporting/Report database.rst [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+Output per. 20 November 2020: 446
+
+Lists all columns in all tables
+-------------------------------
+
+select TABLE_SCHEMA + '.' + TABLE_NAME as ' TableSchema.Table_Name ' , max ( ORDINAL_POSITION ) as ColumnsCount
+from INFORMATION_SCHEMA . COLUMNS where   TABLE _NAME not in ( '__EFMigrationsHistory' , 'Snapshots' , 'PowerBiConfigurations' , 'Commits' ) and TABLE_SCHEMA <> 'Sys'
+group by TABLE_SCHEMA + '.' + TABLE_NAME order by TABLE_SCHEMA + '.' + TABLE_NAME
+
+.. image:: Tables.png
```
Comment (esbenbach) [inline source/reporting/Report database.rst:246]: .. image:: /_images/Tables.png
    :align: center
    :alt: Example list of all accessible tables in the reporting database.

File: source/reporting/Report database.rst [hunk 6]
```diff
@@ -0,0 +1,248 @@
...
+
+select count ( COLUMN_NAME ) CountColumns from INFORMATION_SCHEMA . COLUMNS where TABLE_NAME not in ( '__EFMigrationsHistory' , 'Snapshots' , 'PowerBiConfigurations' , 'Commits' ) and TABLE_SCHEMA <> 'Sys'
+
+
+Output per. 20 November 2020: 446
+
+Lists all columns in all tables
+-------------------------------
+
+select TABLE_SCHEMA + '.' + TABLE_NAME as ' TableSchema.Table_Name ' , max ( ORDINAL_POSITION ) as ColumnsCount
```
Comment (esbenbach) [inline source/reporting/Report database.rst:242]: 1. Are we sure the reporting user has access to the information schema? I rather doubt it. And this sort of selection basically encourages "experimentation" something we don't really want :)
2. The filtered tables should at least be removed and we should attempt to make sure the user does not have access to those tables. If the user gets them while listing tables then so be it, but as long the reporting user can't access them it is not dangerous in any way.

File: source/reporting/Report database.rst [hunk 7]
```diff
@@ -0,0 +1,248 @@
...
+Invoicing of customers produces a demand for payment, and PaymentDemands contains customers' invoices, with amounts, time for when they were issued and when the demand is due. Link to ledger is displayed.
+
+Payment Demand is the basis for payment claims that the system generates, and it is on the basis of these that an invoice is formed. Examples of using payment requirements k an be reporting outstanding guilty amount or income broken down by month of the year. Each claim points to the subscription period that is the basis for the claim. It is important here to keep in mind that the requirement is formed before the period is formed, so the pointer is therefore always backwards in time.  
+
+For orders, there will be a pointer to the order that led to the claim.
+ 
+
+[economy].[SubscriberAmounts]
+-----------------------------
+Information on all amounts paid for a subscriber, divided into different subscriptions.
```
Comment (havmunke) [inline source/reporting/Report database.rst:106]: This is the sum of subscriberledgers grouped by subscriberid, organizationid and currency.

File: source/reporting/Report database.rst [hunk 8]
```diff
@@ -0,0 +1,248 @@
...
+
+ 
+
+[economy].[SubscriberLedgers]
+-----------------------------
+Displays the customer ledger and the types of financial transactions that have been performed for the subscriber.
```
Comment (havmunke) [inline source/reporting/Report database.rst:112]: This is all ledger-entries for all subscribers.

File: source/reporting/Report database.rst [hunk 9]
```diff
@@ -0,0 +1,248 @@
...
+| Balance | Balance / balance adjustment |
++---------+------------------------------+
+| Credit  | Credit                       |
++---------+------------------------------+
+
+
+
+[Order].[OrderAmounts]
+------------------------
+Displays all orders in total (created, canceled and executed) and the value of these. Formed when an order is created and updated when it is completed or canceled.
```
Comment (havmunke) [inline source/reporting/Report database.rst:136]: Grouped by date.

File: source/reporting/Report database.rst [hunk 10]
```diff
@@ -0,0 +1,248 @@
...
+
+ 
+
+[Order].[OrderCompletedAmounts]
+-------------------------------
+Displays the total number of orders executed and the total value of these. The TemplatePackageName field shows which package template was used .
```
Comment (havmunke) [inline source/reporting/Report database.rst:142]: Grouped by date., templatepackage and organization. Only completed orders.

File: source/reporting/Report database.rst [hunk 11]
```diff
@@ -0,0 +1,248 @@
...
+|  1     | Completed   |
++--------+-------------+
+|  2     | Cancelled   |
++--------+-------------+
+
+ 
+
+[Order].[Products]
+------------------
+Displays all products ordered.
```
Comment (havmunke) [inline source/reporting/Report database.rst:164]: Not a standalone table. Always connected to an order-row.

File: source/reporting/Report database.rst [hunk 12]
```diff
@@ -0,0 +1,248 @@
...
+--------------------------------
+General information about framework agreements.
+
+[subscription].[SubscriberAccounts]
+---------------------------------
+Contains the first date for when a SubscriberAccount was used. This is used for financial reporting.
+
+[subscription].[SubscriptionPackageProducts]
+--------------------------------------------
+All subscription packages related to product.
```
Comment (havmunke) [inline source/reporting/Report database.rst:194]: It's rather products included in a subscriptionPackage.

**Review by havmunke — COMMENTED** [review]
Added some comments that should be looked at.

File: source/reporting/Report database.rst (hunk 1 above)
Comment (BrynjarS) [reply source/reporting/Report database.rst:48]: Is this better? Subscriber is a register of all subscribers and contains all contact information from the primary address.

Comment (havmunke) [reply source/reporting/Report database.rst:48]: Yes

File: source/reporting/Report database.rst (hunk 4 above)
Comment (BrynjarS) [reply source/reporting/Report database.rst:66]: Updated and this example part is now removed: "For example, if you agree on a payment claim, which you will later buy out of, then this table will be used."

File: source/reporting/Report database.rst (hunk 6 above)
Comment (BrynjarS) [reply source/reporting/Report database.rst:242]: Yes, just tested - the reporting user has access to the information schema.

File: source/reporting/Report database.rst (hunk 2 above)
Comment (BrynjarS) [reply source/reporting/Report database.rst:61]: Updated!

File: source/reporting/Report database.rst (hunk 3 above)
Comment (BrynjarS) [reply source/reporting/Report database.rst:1]: Updated!

File: source/reporting/Report database.rst (hunk 5 above)
Comment (BrynjarS) [reply source/reporting/Report database.rst:246]: Updated!

File: source/reporting/Report database.rst (hunk 7 above)
Comment (BrynjarS) [reply source/reporting/Report database.rst:106]: Updated

File: source/reporting/Report database.rst (hunk 8 above)
Comment (BrynjarS) [reply source/reporting/Report database.rst:112]: Updated

File: source/reporting/Report database.rst (hunk 9 above)
Comment (BrynjarS) [reply source/reporting/Report database.rst:136]: Updated

File: source/reporting/Report database.rst (hunk 10 above)
Comment (BrynjarS) [reply source/reporting/Report database.rst:142]: Updated

File: source/reporting/Report database.rst (hunk 11 above)
Comment (BrynjarS) [reply source/reporting/Report database.rst:164]: Is this better? Displays products ordered, connected to an order-row.

**Pull request merged**